### PR TITLE
rudrapatel2003/Authenticate server actions and API endpoints

### PR DESCRIPTION
**Pull request by RudraPatel2003**
# Description
This PR makes it so that you must have a valid next-auth cookie to be able to hit a sever endpoint or call specific server actions. 

## Relevant issue(s)

#95 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have assigned reviewers to this PR

### CODE REVIEW

File: src/utils/auth.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { authOptions } from '@/app/api/auth/[...nextauth]/options';
+import { getServerSession } from 'next-auth';
+import { JSendResponse } from './types';
+
+export function authenticateServerAction() {
+  const session = getServerSession(authOptions);
```
Comment (cfisher36) [inline src/utils/auth.ts:6]: needs async/await

**Review by cfisher36 — CHANGES_REQUESTED** [review]

File: src/utils/auth.ts (hunk 1 above)
Comment (RudraPatel2003) [reply src/utils/auth.ts]: 0e51590

File: src/app/api/agencies/route.ts [hunk 2]
```diff
@@ -6,8 +6,11 @@ import {
 } from '@/server/actions/Agencies';
 import '@/server/models/Service';
 import { AgencyInfoForm, JSendResponse } from '@/utils/types';
+import { authenticateServerEndpoint } from '@/utils/auth';
 
 export async function GET() {
+  await authenticateServerEndpoint();
```
[the remark this answers is not in the document]
Comment (RudraPatel2003) [reply src/app/api/agencies/route.ts:12]: 95d05f3 I think does what you mean? What you said makes sense though; good catch on both!

[the remark this answers is not in the document]
Comment (cfisher36) [reply src/app/api/agencies/route.ts]: was a little unclear, but I have included the changes in b70ee50

**Review by cfisher36 — APPROVED** [review]
looks great!

**Pull request merged**